lldb/source/API/SBFrame.cpp

```diff
@@ -1012,33 +1006,26 @@ bool SBFrame::GetDescription(SBStream &description) {
 SBValue SBFrame::EvaluateExpression(const char *expr) {
   LLDB_INSTRUMENT_VA(this, expr);
 
-  SBValue result;
   std::unique_lock<std::recursive_mutex> lock;
   ExecutionContext exe_ctx(m_opaque_sp.get(), lock);
 
   StackFrame *frame = exe_ctx.GetFramePtr();
   Target *target = exe_ctx.GetTargetPtr();
+  SBExpressionOptions options;
   if (frame && target) {
```

Comment:
Should this still check the target? Maybe something like:

```
if(frame) {
  if (Target* target = frame->CalculateTarget()) 
    lldb::DynamicValueType fetch_dynamic_value = target->GetPreferDynamicValue();
    options.SetFetchDynamicValue(fetch_dynamic_value);
  }
}
```

Reply:
Maybe? I'm not a 100% sure which is why I wasn't suggesting it. Even the StackFrame implementation itself seems pretty inconsistent in whether it uses the execution context or `CalculateTarget`. I'd keep the existing behavior to avoid regressing anything.

Reply:
Then I'd leave it as is. Everywhere else it is done in the same way. E.g.
```
  if (frame && target) {
    lldb::DynamicValueType use_dynamic =
        frame->CalculateTarget()->GetPreferDynamicValue();
    sb_value = GetValueForVariablePath(var_path, use_dynamic);
  }
```
or
```
  if (frame && target) {
    lldb::DynamicValueType use_dynamic =
        frame->CalculateTarget()->GetPreferDynamicValue();
    value = FindVariable(name, use_dynamic);
  }
```
and so on.